pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -813,4 +817,22 @@ public TransactionBuilder newTransaction() {
         return new TransactionBuilderImpl(this);
     }
 
+    @SuppressWarnings("unchecked")
+    private static Schema cloneSchema(Schema schema){
```

Comment:
I think we need to implement the `clone` method for each schema implementation. So you can then just call `schema.clone()` to clone a schema.